### DISCUSSION TITLE
Limiter in app "Offline"

### DISCUSSION THREAD

**Discussion by N3llyDeg** [syssi/esphome-soyosource-gtn-virtual-meter#157] [General]
Hey,
the app tells me "Limiter: Offline".

I use the sensor value from HA and from the logs the value is read correctly.

I have tried with and without resistor... any suggestions are appreciated!

Model: 1000W/PV: 26-45V/Battery: 24V

**Comment by Chickenbreast0** [comment]
More information would be appreciated. Logs and config..

**Comment by N3llyDeg** [comment]
Here is the configuration in esphome:

substitutions:
  name: soyosource-gtn-virtual-meter
  device_description: "Monitor a Soyosource GTN and control the power output on demand both via RS485"
  external_components_source: github://syssi/esphome-soyosource-gtn-virtual-meter@main
  tx_pin: GPIO1
  rx_pin: GPIO3

esphome:
  name: soyocontrol
  friendly_name: SoyoControl

esp8266:
  board: d1_mini

external_components:
  - source: ${external_components_source}
    refresh: 0s


logger:
  level: INFO

web_server:
  port: 80

api:

ota:
  password: "xxx"

wifi:
  ssid: "xxx"
  password: "xxx"

  ap:
    ssid: "Soyocontrol Fallback Hotspot"
    password: "xxx"

uart:
  - id: uart_0
    baud_rate: 4800
    tx_pin: ${tx_pin}
    rx_pin: ${rx_pin}

soyosource_modbus:
  - id: modbus0
    uart_id: uart_0
    name: SM

soyosource_virtual_meter:
  - id: virtualmeter0
    soyosource_modbus_id: modbus0

    power_id: powermeter0

    power_sensor_inactivity_timeout: 20s
    power_demand_calculation: NEGATIVE_MEASUREMENTS_REQUIRED
    min_power_demand: 0
    max_power_demand: 300
    zero_output_on_min_power_demand: true
    power_demand_divider: 1
    buffer: 10

    update_interval: 3s

number:
  - platform: soyosource_virtual_meter
    soyosource_virtual_meter_id: virtualmeter0
    buffer:
      name: "${name} buffer"
      initial_value: 10
      restore_value: true
    manual_power_demand:
      name: "${name} manual power demand"
      max_value: 300
    max_power_demand:
      name: "${name} max power demand"
      initial_value: 50
      max_value: 300
      restore_value: true
    power_demand_divider:
      name: "${name} power demand divider"
      initial_value: 1
      restore_value: true

sensor:
  - platform: soyosource_virtual_meter
    soyosource_virtual_meter_id: virtualmeter0
    power_demand:
      name: "${name} power demand"

  - platform: homeassistant
    id: powermeter0
    name: "${name} smartmeter instantaneous power"
    entity_id: sensor.smartmeter_bereinigt
    filters:
      - throttle_average: 15s

switch:
  - platform: soyosource_virtual_meter
    soyosource_virtual_meter_id: virtualmeter0
    manual_mode:
      name: "${name} manual mode"
      restore_mode: RESTORE_DEFAULT_ON
    emergency_power_off:
      name: "${name} emergency power off"
      restore_mode: RESTORE_DEFAULT_OFF

text_sensor:
  - platform: soyosource_virtual_meter
    soyosource_virtual_meter_id: virtualmeter0
    operation_mode:
      name: "${name} limiter operation mode"

**Comment by N3llyDeg** [comment]
This is the hardware configuration I use:
![image](https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/assets/59265369/1b648e42-7b76-4717-9e23-e23fb8361d0a)

**Reply by syssi** [reply]
You missed to setup the flow control pin properly.
